### PR TITLE
Fixes #58 - Auto increment for default name

### LINKED ISSUE #58: default name should auto increment
default name is "worker". if you try to use the default more than once it should be worker-1, worker-2, etc

### PR DESCRIPTION
**Pull request by xprazak2**
Fixes #58 

Stores the name of the last default project
in global config and increments it
when user generates a project without a name.

### CODE REVIEW

**Comment by xprazak2** [comment]
All tests are finally passing. Apologies it took me so long to make them green.

**Comment by ashleymichal** [comment]
hey @xprazak2 !

Sorry this has been sitting on the stack for awhile. I reviewed the code, and actually I don't think we want to cache the directory name every time wrangler runs; rather, we want to have Wrangler respond to an "already exists" OS error by reading the contents of the current working directory, and adding a number if the default name already exists, similar to how filenames are updated in a downloads folder when you download the same asset twice.

So, if `worker` already exists in the CWD, create `worker2`; if `worker2` already exists, create `worker3`, etc.

We really try to minimize the amount of local state Wrangler has to keep track of to keep our footprint relatively small.

This does have a small disadvantage, wherein one could accidentally overwrite a worker because of naming collisions, but we would rather solve that with #330 .

**Comment by xprazak2** [comment]
Thank you for your review, I have made changes as suggested - wrangler no longer keeps additional state and scans current dir for the next available name.

File: src/commands/generate/mod.rs [hunk 1]
```diff
@@ -33,14 +35,80 @@ pub fn run_generate(name: &str, template: &str) -> Result<(), failure::Error> {
...
+        num = num.map(|val| val + 1);
+        new_name = construct_name(&bare_name, num);
+        if num.is_none() {
+            num = Some(1);
+        }
+    }
+    Ok(new_name)
+}
+
+fn read_current_dir() -> Result<Vec<OsString>, failure::Error> {
```
Comment (ashleymichal) [inline src/commands/generate/mod.rs]: i would also suggest passing the name in and only returning colliding directories.

**Comment by ashleymichal** [comment]
hey @xprazak2 , just bumping this, sorry i forgot to mention you in my review last week. what do you think about these comments? if you have the bandwidth to implement that would be lovely, if not please let me know and i can pick this up and finish it up, you'll of course be credited in the changelog. thanks again for this PR, it is very  much appreciated!

**Comment by xprazak2** [comment]
I simplified the logic for generating a new name. The project name is no longer scanned for digits at the end but `-\d+` is appended when we detect the dir with the same name already exists.

**Comment by xprazak2** [comment]
@ashleymichal, any chance you could do a re-review please?

**Pull request merged**